### DISCUSSION TITLE
Rounded corners for borders?

### DISCUSSION THREAD

**Discussion by EZSWTool01** [QuestPDF/QuestPDF#427] [Q&A] [answered]
Hi. I believe someone else has asked a similar question (Radius). Is it possible to present rounded corners for the borders?

**Comment by MarcinZiabek** [comment, marked as answer]
Yes, it is possible. You can use the `Canvas` element to draw any content you like, including rectangles with rounded corners. There is a special API method available in SkiaSharp for this type of tasks.

I am investigating adding this capability in one of the future releases.

**Comment by EZSWTool01** [comment]
Thank you very much!

**Comment by ahmedbou2000** [comment]
//this is my methode to create and controle each border in the rectangle :
public static void DrawRoundedRectangle(SKCanvas canvas,
    QuestPDF.Infrastructure.Size size,
    float leftRadius =0,
    float topRadius = 0,
    float rightRadius = 0,
    float bottomRadius = 0,
    string color = "#FFFFFF"
    )
{

    float width = size.Width;
    float height = size.Height;
    int x = 0;
    int y = 0;

    SKRoundRect rrect = new SKRoundRect();
    SKPoint[] points = new SKPoint[] {new SKPoint(leftRadius, topRadius),
                                    new SKPoint(rightRadius, topRadius),
                                    new SKPoint(rightRadius, bottomRadius),
                                    new SKPoint(leftRadius, bottomRadius)};
    rrect.SetRectRadii(new SKRect(x, y, x + width, y + height), points);

    SKPaint paint = new SKPaint
    {
        IsAntialias = true,
        IsStroke = true,
        Color = SKColor.Parse(color)
    };

    canvas.DrawRoundRect(rrect, paint);
}

// this is how you can use it :

page.Content()
.Background(Colors.Grey.Lighten3)
.Column(contentColumn =>
{
    contentColumn.Spacing(3);

    contentColumn.Item()
    .Layers(layers =>
    {
        layers.Layer().Canvas((canvas, size) =>
        {
            Methods.DrawRoundedRectangle(canvas, size,topRadius:10f,leftRadius:10f,rightRadius:10 ,color:"#000000");
        });

        layers.PrimaryLayer().Height(100);
    });
});

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Since this discussion was opened, the `.Canvas()` method has been deprecated. For drawing shapes and lines, please refer to the [Line](https://www.questpdf.com/api-reference/line.html), [SVG](https://www.questpdf.com/api-reference/svg.html), and [Complex Graphics](https://www.questpdf.com/api-reference/complex-graphics.html) documentation.

Should you encounter any further problems, please don't hesitate to open a new discussion with additional context.

Thank you!
